code-challenges.test.js

```diff
@@ -11,41 +11,113 @@
+// --------------------1) Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
 
-// Pseudo code:
 
+// Function to check if a number inside an object is evenly divisible by three
+// const checkDivisibileByThree = (divisible) => {
+//   const { number } = divisible
+//   if (number % 3 === 0) {
+//     return `${number} is divisible by 3`
+//   } else {
+//     return `${number} is not divisible by 3`
+//   }
+// }
+
+// // jest test using describe
+// describe('Check Divisibile By Three', () => {
+//   const object1 = { number: 15 }
+//   // Expected output: "15 is divisible by three"
+//   const object2 = { number: 0 }
+//   // Expected output: "0 is divisible by three"
+//   const object3 = { number: -7 }
+//   // Expected output: "-7 is not divisible by three"
+// // jest test using it
+//   it('should return "15 is divisible by three"', () => {
+//     //jest test using expect with
+//       expect(checkDivisibileByThree(object1)).toBe("15 is divisible by 3")
+//   })
+
+//   it('should return "0 is divisible by three"', () => {
+//       expect(checkDivisibileByThree(object2)).toBe("0 is divisible by 3")
+//   })
+
+//   it('should return "-7 is not divisible by three"', () => {
+//       expect(checkDivisibileByThree(object3)).toBe("-7 is not divisible by 3")
+//   })
+// })
+
+// // Pseudo code:
+// // a) Create a test with expect statements for each of the variables provided.
+// // b) Create the function that makes the test pass.
+// // c) Create the anatomy of a Jest Test with describe statement to set a parameter
+// // d) Create a -it- Jest Test to inform us what the describe statement should be doing
+// e) Create an -expect- Jest Test to have a argument to invoke
+  
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
-// a) Create a test with expect statements for each of the variables provided.
+// Pseudo code:
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
-// b) Create the function that makes the test pass.
+// a) Create a test with expect statements for each of the variables provided.
+// Function to capitalize all words in an array
+function capitalizeWords(wordArray) {
+  // .map to use on all values of the array
+  // .charAt(0) to pinpoint the first letter and index of each string
+  // .toUppercase() once we are able to access each string from the array we can target the first letter with .charAt to capitaliaze it
+  //  .slice() method returns selected elements in an array as a new array
+  return wordArray.map(word => word.charAt(0).toUpperCase() + word.slice(1))
+}
+
```

Comment:
You are not using slice on an array like you said in your pseudo code. The code is applying .slice on each string.